modules/videojsVideoProvider.js

```diff
@@ -25,29 +27,82 @@ export function VideojsProvider(config, videojs_, adState_, timeState_, callback
 
     // returns the player if it exists, or attempts to instantiate a new one
     player = videojs(divId, playerConfig, function() {
-        // callback runs in both cases
+      // callback runs in both cases
     });
+    // TODO: make sure ortb gets called in integration example
+    // currently testing with a hacky solution by hooking it to window
+    // window.ortb = this.getOrtbParams
   }
 
   function getId() {
     return divId;
   }
 
   function getOrtbParams() {
+    if (!player) {
+      return null;
+    }
+
+    let playBackMethod = PLAYBACK_METHODS.CLICK_TO_PLAY
+    const isMuted = player.muted() || player.autoplay() === 'muted'; // todo autoplayAdsMuted only applies to preRoll
+    if (player.autoplay()) {
+      playBackMethod = isMuted ? PLAYBACK_METHODS.AUTOPLAY_MUTED : PLAYBACK_METHODS.AUTOPLAY;
+    }
+    const supportedMediaTypes = Object.values(VIDEO_MIME_TYPE).filter(
+      // Follows w3 spec https://www.w3.org/TR/2011/WD-html5-20110113/video.html#dom-navigator-canplaytype
+      type => player.canPlayType(type) !== ''
+    ).concat([VPAID_MIME_TYPE])
     const video = {
-      mimes: [],
-      w: 0,
-      h: 0,
+      mimes: supportedMediaTypes,
+      // Based on the protocol support provided by the videojs-ima plugin
+      // https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/compatibility
+      // Need to check for the plugins prescence by checking videojs.ima
+      protocols: videojs.ima ? [
+        PROTOCOLS.VAST_2_0,
+      ] : [],
+      api: [
+        API_FRAMEWORKS.VPAID_2_0
+      ],
+      // TODO: Make sure this returns dimensions in DIPS
+      h: player.currentHeight(),
+      w: player.currentWidth(),
+      placement: PLACEMENT.IN_STREAM,
```

Comment:
could video.js ever be used only as an ads player ? If so it would be `outstream`. This distinction might also be missing in the jwplayer submodule since we do support an ads only mode.